### PR TITLE
Fixing issue were windows PR job has duplicate name for 1.28 branch

### PR DESCRIPTION
**Pull request by marosset**
Because the job name didn't have `master` in it the job got forked as part of setting up the v1.28 release branches but the jobs did not get a unique name.

This PR fixes but the issue with the job forking and the release-1.28 job with a duplicate name

/sig windows
/assign @jsturtevant @jeremyrickard

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml [hunk 1]
```diff
@@ -2022,7 +2022,7 @@ presubmits:
   - always_run: false
     branches:
     - release-1.28
-    context: pull-kubernetes-e2e-capz-windows
+    context: pull-kubernetes-e2e-capz-windows-1.28
```
Comment (jsturtevant) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml:2025]: I am not familiar with the value, what is this for?

Comment (marosset) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml:2025]: i'm not sure... maybe it is the job that the forked job was created from?
will look...

Comment (marosset) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml]: updated

File: config/jobs/kubernetes-sigs/sig-windows/release-1.28-windows-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+presubmits:
+  kubernetes/kubernetes:
+  - name: pull-kubernetes-e2e-capz-windows-1.28
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.28-windows-presubmits.yaml:3]:
```suggestion
  - name: pull-kubernetes-e2e-capz-windows-1-27
```

to pass CI and match other jobs

Comment (marosset) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.28-windows-presubmits.yaml:3]: updated

**Pull request merged**